### PR TITLE
Feature(state): MerkleSchedule

### LINKED ISSUE #72: Scheduled Entity State
New FCMap will be added (`Map<MerkleEntityId, MerkleSchedule>`).
`MerkleSchedule` will store information related to scheduled transactions:

- `transactionBody`
- `schedulingAccount`
- `signKey` <-- to discuss with Michael and Atul
- `sigMap`

### PR DESCRIPTION
**Pull request by failfmi**
**Related issue(s)**:
Closes #72 

**Summary of the change**:
* Add MerkleSchedule
* Add new FCMap<MerkleEntityId, MerkleSchedule>

**External impacts**:
None.

**Applicable documentation**
- [ ] Javadoc
- [ ] HAPI protobuf comments
- [ ] README

### CODE REVIEW

**Comment by failfmi** [comment]
We can discuess whether naming should be Scheduled or Schedule

File: hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleSchedule.java [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    static final long RUNTIME_CONSTRUCTABLE_ID = 0x8d2b7d9e673285fcL;
+    static DomainSerdes serdes = new DomainSerdes();
+
+    public static final JKey UNUSED_KEY = null;
+
+    private byte[] transactionBody;
+    private JKey adminKey = UNUSED_KEY;
+    private HashSet<EntityId> signers = new HashSet<>();
+    private Map<EntityId, byte[]> signatures = new HashMap<>();
+    private boolean executedImmediately;
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleSchedule.java:63]: I think that the name of this property should be `executeImmediately`

File: hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleSchedule.java [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+            return false;
+        }
+
+        var aKeys = a.keySet();
+        var bKeys = b.keySet();
+        var iterator = bKeys.iterator();
+
+        for (EntityId aKey : aKeys) {
+            var bKey = iterator.next();
+            if (!aKey.equals(bKey)) {
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleSchedule.java:288]: .

File: hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleSchedule.java [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+        return RUNTIME_CONSTRUCTABLE_ID;
+    }
+
+    @Override
+    public int getVersion() {
+        return MERKLE_VERSION;
+    }
+
+    @Override
+    public FCMElement copy() {
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleSchedule.java:164]: Shouldn't this be `MerkleSchedule`

File: hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleSchedule.java [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+    }
+
+    public boolean isExecutedImmediately() { return executedImmediately; }
+
+    public void setExecutedImmediately(boolean executedImmediately) { this.executedImmediately = executedImmediately; }
+
+    public Map<EntityId, byte[]> signatures() { return signatures; }
+
+    public void putSignature(EntityId key, byte[] signature) {
+        if (signature.length != 64) {
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleSchedule.java:216]: you could use  `SIGNATURE_BYTES` here

File: hedera-node/src/test/java/com/hedera/services/state/merkle/MerkleScheduleTest.java [hunk 5]
```diff
@@ -0,0 +1,438 @@
...
+    }
+
+    @Test
+    public void failDifferentExecutedImmediately() {
+        // given:
+        other = new MerkleSchedule(transactionBody, signers, signatures);
+        setOptionalElements(other);
+
+        // when:
+        other.setDeleted(otherExecutedImmediately);
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/test/java/com/hedera/services/state/merkle/MerkleScheduleTest.java:293]: should be `other.setExecutedImmediately`

**Review by Daniel-K-Ivanov — APPROVED** [review]

**Pull request merged**